crates/invoker-impl/src/lib.rs

```diff
@@ -354,7 +355,9 @@ where
                     .registered_partitions_with_keys(keys.clone())
                     .flat_map(|partition| self.status_store.status_for_partition(partition))
                     .filter(|status| keys.contains(&status.invocation_id().partition_key()))
+                    .sorted_by_key(|e| e.invocation_id().partition_key())
```

Comment:
What happens if we let DF handle the sorting of the invocation state table? Would this lead to sub-optimal execution plans?